### DISCUSSION TITLE
fluent-bit critical vulnerabilty not detected

### DISCUSSION THREAD

**Discussion by pradyumnaparalikar** [aquasecurity/trivy#6793] [Q&A] [answered]
### IDs

CVE-2024-4323

### Description

`fluent-bit` component for log processing. There was CRITICAL 9.8 CVE discovered in the application recently - https://fluentbit.io/blog/2024/05/21/statement-on-cve-2024-4323-and-its-fix/
 
The problem is that trivy isn’t detecting `fluent-bit` being installed in the container image and thus not reporting the issue. This problem can be easily replicated by running trivy scanner locally:

### Reproduction Steps

```bash
trivy i cr.fluentbit.io/fluent/fluent-bit:2.1.4 -s CRITICAL
```

### Target

Container Image

### Scanner

Vulnerability

### Debug Output

```bash
trivy i cr.fluentbit.io/fluent/fluent-bit:2.1.4 -s CRITICAL
```

### Version

```bash
Version:v0.44.0
```

### Checklist

- [X] Read [the documentation regarding wrong detection](https://aquasecurity.github.io/trivy/dev/community/contribute/discussion/#false-detection)
- [ ] Ran Trivy with `-f json` that shows data sources and confirmed that the security advisory in data sources was correct

**Comment by DmitriyLewen** [comment, marked as answer]
Hello @pradyumnaparalikar 
Thanks for your interest to Trivy!

`fluent-bit` was installed in `cr.fluentbit.io/fluent/fluent-bit:2.1.4` image using `COPY` (this means that we can't classify it as a package manager(apt, apk, etc.):
```bash
➜ docker history cr.fluentbit.io/fluent/fluent-bit:2.1.4
IMAGE          CREATED         CREATED BY                                      SIZE      COMMENT
...
<missing>      12 months ago   COPY /fluent-bit /fluent-bit # buildkit         46.7MB    buildkit.dockerfile.v0
```

Trivy supports only `Binaries built by Go` and `Binaries built with cargo-auditable` (see https://aquasecurity.github.io/trivy/v0.51/docs/coverage/language/).

That is why Trivy doesn't detect `fluent-bit` and `CVE-2024-4323`.

Regards, Dmitriy

**Reply by pradyumnaparalikar** [reply]
Thank you for the update @DmitriyLewen
